### PR TITLE
Review branch

### PR DESCRIPTION
**Pull request by jordaninthewind**
Updated status codes and error handling.

### CODE REVIEW

**Comment by MurDaD** [comment]
Still unable to create user.
Payload:
```
{
	"login": "TestLogin",
	"password": "asdasdasd",
	"age": 22
}
```
Error:
`Error validating request body. "login" is required. "password" is required. "age" is required.`

Also error should be JSON

**Comment by MurDaD** [comment]
When user is created, the response should contain only user ID (not required, but common). You can skip that if you want

**Comment by MurDaD** [comment]
Minor: Can create user with the same user ID (Please note that when doing 3rd task - you should not show DB error as a response)

**Comment by MurDaD** [comment]
Data response should not return all the fields in `data` property - all the return fields should be in top level (exception - when you're using AWS API Gateway)

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+
+import router from './routes/routes';
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (MurDaD) [inline src/index.js]: You need only `app.use(bodyParser.json());` - urlencoded is used for multipart form data objects

Comment (jordaninthewind) [reply src/index.js]: So, are you sending the query JSON in the headers with the Content-Type as application/json? I can't get anything to show up in the req.body if I change that to bodyParser.json() and add it in headers. Noted on the use of urlencoded.

Comment (jordaninthewind) [reply src/index.js]: This took an embarrassingly long amount of time to work out how to send this correctly in Postman. Ah man.

**Comment by jordaninthewind** [comment]
Updated body-parser usage. For putting all fields in top level for JSON response, does this mean that I should spread the response into JSON? For instance, if there are multiple users, should they each have a key? And finally, for the error response, it's the Joi middleware that's handling that. I'll look into how to update (there's possibly something built in), but at this point it's sending that because it's its own default behavior.

**Comment by jordaninthewind** [comment]
Fixed error response format.

**Pull request closed without merging**